src/core/api.ts

```diff
@@ -45,8 +49,23 @@ export function createApi(
     hg: Readonly<HiGlassApi>,
     hgSpec: HiGlassSpec | undefined,
     tracksAndViews: readonly VisUnitApiData[],
-    theme: Required<CompleteThemeDeep>
+    theme: Required<CompleteThemeDeep>,
+    idTable: Readonly<IdTable>
 ): GoslingApi {
+    const idTableCopy = JSON.parse(JSON.stringify(idTable));
```

Comment:
```suggestion
    const idTableCopy = structuredClone(idTable);
```
consider using structuredClone() instead